hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/utils/SystemContractUtils.java

```diff
@@ -73,6 +73,17 @@ public static ContractFunctionResult contractFunctionResultSuccessFor(
                 .build();
     }
 
+    @NonNull
+    public static ContractFunctionResult contractFunctionResultSuccessFor(
+            final long gasUsed, final Bytes result, AccountID senderId) {
+        return ContractFunctionResult.newBuilder()
+                .gasUsed(gasUsed)
+                .contractCallResult(tuweniToPbjBytes(result))
+                .senderId(senderId)
+                .contractID(HTS_PRECOMPILE_MIRROR_ID)
+                .build();
+    }
```

Comment:
Just FYI @natanasow @JivkoKelchev - you can have some merge conflicts here https://github.com/hashgraph/hedera-services/pull/10977/files#diff-50bf7a98ab68915e3012c2d296c58315119bc2aca16f6054d74cea7c84db2dd3R85